velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -28,51 +28,117 @@ namespace facebook::velox::exec::prefixsort {
 /// Provides encode/decode methods for PrefixSort.
 class PrefixSortEncoder {
  public:
-  /// 1. Only int64_t is supported now.
-  /// 2. Encoding is compatible with sorting ascending with no nulls.
+  /// Encode native primitive types(such as byte, int16, int32, int64, double,
+  /// float) to dest. There will be a separate method to handle string type as
+  /// follow-up.
+  /// 1. The first byte of the encode result is nullByte, which is determined by
+  ///    the nullsFirst :
+  ///    -If nullsFirst : invalid(means null) -> 0, valid(means not null) -> 1
+  ///    -If !nullsFirst : invalid(means null) -> 1, valid(means not null) -> 0
+  /// 2. The remaining bytes are the encoding result of value:
+  ///    -If isNull, we ignore the input value (this means the caller
+  ///     can pass any value, the reason for not using std::optional here is to
```

Comment:
What is this overhead? What do you see in the benchmark?

Reply:
```
char* buffer1[5];
char* buffer2[5];
int64_t v = 100L;

BENCHMARK(encodeOptianal) {
  for (unsigned long i = 0; 0xFFFFFFFF; i++) {
    for (unsigned long j = 0; j < 0xFFFFFFFF; ++j) {
      prefixsort::PrefixSortEncoder::encode(
          true, true, false, (int64_t )(i + j), (char*)buffer1);
    }
  }
}

BENCHMARK(encode) {
  long v = 1;
  for (unsigned long i = 0; i < 0xFFFFFFFF; i++) {
    for (unsigned long j = 0; j < 0xFFFFFFFF; ++j) {
      prefixsort::PrefixSortEncoder::testEncode(
          true, true, std::optional((int64_t )(i + j)), (char*)buffer2);
    }
  }
}
```

```
============================================================================
[...]arks/PrefixSortAlgorithmBenchmark.cpp     relative  time/iter   iters/s
============================================================================
encodeOptianal                                              0.00fs  Infinity
encode                                                      0.00fs  Infinity
```

I build a small benchmark , seems cost nothing, I will change it to std::optional

Reply:
Thanks.